### PR TITLE
tunnel: add source mode proxy

### PR DESCRIPTION
**Pull request by at-wat**
For #86

### CODE REVIEW

**Comment by at-wat** [comment]
@kt3k @kamatama41 please take a look

File: tunnel/proxy.go [hunk 1]
```diff
@@ -27,149 +23,86 @@ func endpointHost(region string) string {
 // Dialer is a proxy destination dialer.
 type Dialer func() (io.ReadWriteCloser, error)
 
-func (t *tunnel) proxy(ctx context.Context, dialer Dialer, notification *notification, opts ...proxyOption) error {
-	if notification.ClientMode != Destination {
-		return errors.New("unsupported client mode")
+// ProxyDestination local connection via IoT secure tunneling.
```
Comment (kt3k) [inline tunnel/proxy.go:26]: I think the document should mention about "destination device" concept. I guess this method is supposed to be called only on destination device and `ProxySource` is supposed to be called only on "source device". I think these differences are worth mentioning here.

**Comment by kt3k** [comment]
Otherwise it looks good to me.

File: tunnel/proxy.go (hunk 1 above)
Comment (at-wat) [reply tunnel/proxy.go]: Updated the document

File: tunnel/proxy.go [hunk 2]
```diff
@@ -27,149 +23,90 @@ func endpointHost(region string) string {
 // Dialer is a proxy destination dialer.
 type Dialer func() (io.ReadWriteCloser, error)
 
-func (t *tunnel) proxy(ctx context.Context, dialer Dialer, notification *notification, opts ...proxyOption) error {
-	if notification.ClientMode != Destination {
-		return errors.New("unsupported client mode")
+// ProxyDestination proxies TCP connection from remote source device to
+// the local destination application via IoT secure tunneling.
+// This is usually used on IoT things.
```
Comment (kt3k) [inline tunnel/proxy.go:28]: 👍

**Pull request merged**